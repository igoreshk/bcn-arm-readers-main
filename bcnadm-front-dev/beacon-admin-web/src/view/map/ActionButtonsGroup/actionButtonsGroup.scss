.actionButtonsGroup {
  display: flex;
  margin-left: 36px;

  .activeEditButtonIcon, .inactiveEditButtonIcon {
    width: 24px;
    height: 24px;
  }

  .activeEditButtonIcon {
    fill: #ffffff;
  }

  .inactiveEditButtonIcon {
    fill: #B22746;
  }
}