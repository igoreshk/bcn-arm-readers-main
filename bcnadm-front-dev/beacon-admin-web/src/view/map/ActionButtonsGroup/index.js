import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import Edit from '@material-ui/icons/Edit';
import PropTypes from 'prop-types';

import ModeButton, { buttonType } from '../ModeButton';
import ConfirmDeleteDialog from '../ConfirmDeleteDialog';

import './actionButtonsGroup.scss';

class ActionButtonsGroup extends Component {
  render() {
    const { clear, translate, toEdit } = this.props;
    const { edit: editMode } = this.props.match.params;

    function setClassName() {
      return editMode ? 'activeEditButtonIcon' : 'inactiveEditButtonIcon';
    }

    return (
      <div className="actionButtonsGroup">
        <ModeButton
          icon={<Edit className={setClassName()} />}
          onClick={toEdit}
          active={Boolean(editMode)}
          tooltip={translate('operator.edit')}
          buttonId="edit"
          buttonType={buttonType.ACTION}
        />
        {editMode && <ConfirmDeleteDialog clear={clear} translate={translate} />}
      </div>
    );
  }
}

ActionButtonsGroup.propTypes = {
  translate: PropTypes.func.isRequired,
  clear: PropTypes.func.isRequired,
  match: PropTypes.object,
  toEdit: PropTypes.func
};

export default withRouter(ActionButtonsGroup);
