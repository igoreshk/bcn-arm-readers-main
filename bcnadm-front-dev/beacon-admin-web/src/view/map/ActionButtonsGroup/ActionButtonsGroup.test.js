import React from 'react';
import { shallow } from 'enzyme';

import ActionButtonsGroup from './index';

const props = {
  translate: jest.fn(),
  clear: jest.fn()
};

const wrapper = shallow(<ActionButtonsGroup {...props} />);

describe('mock', () => {
  it('render component', () => expect(wrapper.length).toEqual(1));
  it('snapshot for ActionButtonsGroup', () => expect(wrapper).toMatchSnapshot());
});
