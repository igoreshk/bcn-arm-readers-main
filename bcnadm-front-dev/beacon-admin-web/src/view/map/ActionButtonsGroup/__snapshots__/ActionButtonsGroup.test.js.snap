// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock snapshot for ActionButtonsGroup 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
